### DISCUSSION TITLE
Problems with dynamic

### DISCUSSION THREAD

**Discussion by bryan-alvarado-r** [apache/airflow#32853] [General]
Greetings community: I have the next code. of airflow but it doesnt render on ui.

`import os
from datetime import datetime
import sys

from airflow import DAG
from airflow.models import Variable
from airflow.models.baseoperator import chain
from airflow.operators.empty import EmptyOperator
from airflow.operators.trigger_dagrun import TriggerDagRunOperator
from rappiflow.operators import MultiStatementSnowflakeOperator
from settings import get_default_args_opsgenie
from airflow.decorators import dag
from airflow.utils.dag_parsing_context import get_parsing_context

queries_base_path = os.path.join(os.path.dirname(__file__), "queries")

countries = ["CO", "MX", "BR", "AR", "CL", "UY", "PE", "CR", "EC"]

days_dnp_drt = 600
split_steps_int = 100
total_task_in_line_int = split_steps_int - 1
dag_id_lst = []
dag_lst_dct = {}
dag_values_dct = {}

settings = {
    "snowflake_conn_id": "fin_conn_snowflake_repro",
    "google_sheets_id": "conn_google_sheets",
}

default_args = get_default_args_opsgenie(
    vertical, url_notifications, priority, owner_process
)

ENV = Variable.get("ENV", "")

current_dag_id = get_parsing_context().dag_id
print(get_parsing_context())
current_dag = None
if len(sys.argv) > 3:
    print(sys.argv)
    current_dag = sys.argv[3]

for ope in ["DNP", "DRT"]:
    dag_lst_dct[ope] = {}
    for country in countries:
        temp_lst = []
        for i in range(days_dnp_drt, 1, -split_steps_int):
            dag_id_str = "FIN_UE_{}_{}_{}_{}".format(
                ope, country, str(i), str(i - total_task_in_line_int)
            )
            dag_values_dct[dag_id_str] = {
                "start": i,
                "stop": i - total_task_in_line_int,
            }
            temp_lst.append(dag_id_str)
        dag_lst_dct[ope].update({country: temp_lst})

for ope in ["DNP", "DRT"]:
    for country in countries:
        for dag_id in dag_lst_dct[ope][country]:
            if current_dag is not None and current_dag != dag_id:
                continue
            @dag(
                dag_id=dag_id,
                schedule_interval=None,
                start_date=datetime(2020, 1, 14),
                max_active_runs=1,
                default_args=default_args,
                catchup=False,
                template_searchpath=queries_base_path,
                tags=["REPROCESO"],
            )
            def create_dag_by_country():
                start_task = EmptyOperator(
                    task_id="START_TASK", trigger_rule="all_done"
                )
                end_task = EmptyOperator(task_id="END_TASK", trigger_rule="all_done")
                ope_lst = [
                    MultiStatementSnowflakeOperator(
                        task_id="UE_" + ope + "_" + country + "_" + str(day),
                        sql="FIN_UE/FIN_UE_DNP_DRT/TBL_UE_{}.sql".format(ope),
                        snowflake_conn_id=settings["snowflake_conn_id"],
                        params={
                            "country": country,
                            "db": ENV,
                            "schema": "GLOBAL_FINANCES",
                            "months_dnp_drt": day,
                        },
                        trigger_rule="all_done",
                    )
                    for day in range(
                        dag_values_dct[dag_id]["start"],
                        dag_values_dct[dag_id]["stop"] - 1,
                        -1,
                    )
                ]
                chain(*ope_lst)
                start_task >> ope_lst[0]
                ope_lst[-1] >> end_task
create_dag_by_country()`

can you help me or explain me why didn't work please?

**Comment by potiuk** [comment]
Friendly advice: If you look for help of people here (that do it in their free time) then I suggest you at least take care about formatting your message so that it reflects your code. Learning how to do that in GH discussions, is the first step. Not doing it is a sign that you are not willing to spend time to make it easy for others to help to solve your problems. It's a bit strange to ask for help and make it hard for those who might want to spend their time on helping you. There are multiple reasons I might think you would do that, none of those reasons make it likely that somoene would like to spend their time if you have not spent your time. 

Secondly - debugging your code by posting it to discussions is not necessarily what people would like to do - likely - if you ask for help you should show what you've done to solve and investigate your problem and ask a concrete question about the code you had problem with understanding. This is what might make people want to help you - when you see you've done your part. Posting a code and asking "what's wrong" might mean a lot of things. but the main assumption is that you have done nothing and that you just one someone else to debug your code and that you've done nothing to do so. But maybe it is a wrong assumption. But not describing your efforts so far opens up for assumptions like that.

So my friendly advice - if you want people to help you, show that you've done your part and make at least **some** effort to make your questions to be concrete and that your problem is at least formatted well.
